usr/src/cmd/bhyve/block_if.c

```diff
@@ -566,7 +577,7 @@ blockif_open(const char *optstr, const char *ident)
 			ioctl(fd, DIOCGSTRIPEOFFSET, &psectoff);
 		strlcpy(arg.name, "GEOM::candelete", sizeof(arg.name));
 		arg.len = sizeof(arg.value.i);
-		if (ioctl(fd, DIOCGATTR, &arg) == 0)
+		if (nodelete == 0 && ioctl(fd, DIOCGATTR, &arg) == 0)
```

Comment:
This is within `#ifdef __FreeBSD__`.  How does `candelete` ever get set on SmartOS?

Reply:
The second PR enables all of this on illumos (see PR description) -- this represents the bits that we'll plan to upstream.